src/modules/scraper/adapter/messaging/RectifyStickyReferralConsumer.ts

```diff
@@ -0,0 +1,47 @@
+import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+import { IsNull, MoreThan, Not, Repository } from "typeorm";
+import { InjectRepository } from "@nestjs/typeorm";
+
+import { DepositReferralQueueMessage, RectifyStickyReferralQueueMessage, ScraperQueue } from ".";
+import { Deposit } from "../../../deposit/model/deposit.entity";
+import { ScraperQueuesService } from "../../service/ScraperQueuesService";
+
+@Processor(ScraperQueue.RectifyStickyReferral)
+export class RectifyStickyReferralConsumer {
+  private logger = new Logger(RectifyStickyReferralConsumer.name);
+
+  constructor(
+    @InjectRepository(Deposit) readonly depositRepository: Repository<Deposit>,
+    private scraperQueuesService: ScraperQueuesService,
+  ) {}
+
+  @Process({ concurrency: 1 })
+  private async process(job: Job<RectifyStickyReferralQueueMessage>) {
+    const { depositId } = job.data;
+    const deposit = await this.depositRepository.findOne({ where: { id: depositId } });
+
+    if (!deposit) return;
```

Comment:
Similar question: how would this ever happen? And why isn't it a problem?

Reply:
This will happen if an invalid deposit is manually submitted to the `RectifyStickyReferral` queue (for example, by using an admin endpoint). In this case we don't want to retry the message indefinitely